src/AndroidDebugLauncher/NdkReleaseId.cs

```diff
@@ -49,6 +49,89 @@ public bool IsValid
+        /// <param name="result"></param>
+        /// <returns>true if successful</returns>
+        public static bool TryParsePropertiesFile(string ndkSourcePropertiesFilePath, out NdkReleaseId result)
+        {
+            result = new NdkReleaseId();
+
+            using (StreamReader reader = File.OpenText(ndkSourcePropertiesFilePath))
+            {
+                Dictionary<string, string> properties = new Dictionary<string, string>();
+                while (true)
+                {
+                    string line = reader.ReadLine();
+                    if (line == null)
+                        break; // end of file
+
+                    line = line.Trim();
+                    if (line.Length == 0)
+                        continue; // ignore any blank lines. I don't expect there to be any, but it seems reasonable to do
+
+                    // .properties files can theoretically have '=' characters in the value portion,
+                    // but I don't expect this will happen and our parsing logic below can't handle it anyway
+                    string[] keyValue = line.Split('=');
+                    if (keyValue.Length != 2)
+                    {
+                        return false;
+                    }
+
+                    properties.Add(keyValue[0].Trim(), keyValue[1].Trim());
+                }
+
+                string revision;
+                if (properties.TryGetValue("Pkg.Revision", out revision))
+                {
+                    return TryParseRevision(revision, out result);
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Try to parse the revision from the ndk
+        /// </summary>
+        /// <param name="revision">[Required] revision of the ndk in the format major.hotfix.build</param>
+        /// <param name="result">On success, a valid version</param>
+        /// <returns>true if successful</returns>
+        public static bool TryParseRevision(string revision, out NdkReleaseId result)
+        {
+            result = new NdkReleaseId();
+
+            string[] numbers = revision.Split('.');
+            if (numbers.Length != 3)
+            {
+                return false;
+            }
+
+            string releaseString = numbers[0];
+            uint releaseNumber;
+
+            if (!uint.TryParse(releaseString, NumberStyles.None, CultureInfo.InvariantCulture, out releaseNumber) ||
+                releaseNumber == 0)
+            {
+                return false;
+            }
+
+            string subReleaseString = numbers[1];
+            char subReleaseChar = 'a';
+            int subReleaseNumber;
+            if (!int.TryParse(subReleaseString, out subReleaseNumber))
+            {
+                return false;
+            }
+
+            subReleaseChar += Convert.ToChar(subReleaseNumber);
+
+            result = new NdkReleaseId(releaseNumber, subReleaseChar, true);
```

Comment:
The new source.properties file doesn't have any indication of whether it's a 32 or 64-bit NDK.

As far as I can tell, the `Is32Bit` property is only used in the `toString()` method of `NdkReleaseId`.

Let me know if knowing 32 vs 64 is needed, and then we can look into exploring the directory structure of the NDK to figure it out.